packages/playground/src/weblets/tf_contracts_list.vue

```diff
@@ -329,6 +307,76 @@ async function _normalizeContracts(
   return normalizedContracts.filter(Boolean) as NormalizedContract[];
 }
 
+function parseProjectName(projectName: string) {
+  const parts = projectName.split("/");
+  if (parts.length) {
+    projectName = solutionType[parts[0]];
+  }
+  return projectName;
+}
+
+async function normalizeContractWithConsumptionCache(
+  grid: GridClient,
+  c: { [key: string]: any },
+  type: ContractType.Node | ContractType.Name | ContractType.Rent,
+): Promise<NormalizedContract> {
+  const id = +c.contract_id;
+
+  let data: { [key: string]: string };
+  try {
+    data = JSON.parse(c.details.deployment_data);
+    data.projectName = parseProjectName(data.projectName);
+  } catch {
+    data = { name: c.details.name };
+  }
+
+  let expiration = "-";
+  if (c.state === ContractState.GracePeriod) {
+    const exp = await grid.contracts.getDeletionTime({ id });
+    expiration = new Date(exp).toLocaleString();
+  }
+
+  let consumption = { amountBilled: 0, discountReceived: "None" as any };
+  const cached = consumptionCache.value.get(id);
+  if (cached) {
+    consumption = { amountBilled: cached.consumption, discountReceived: cached.discountPackage };
+  } else {
+    try {
+      consumption = await grid.contracts.getConsumption({ id });
+      consumptionCache.value.set(id, {
+        consumption: consumption.amountBilled,
+        discountPackage: consumption.discountReceived,
+      });
+    } catch {
+      consumptionCache.value.set(id, {
+        consumption: 0,
+        discountPackage: "None",
+      });
+    }
+  }
```

Comment:
i have many concerns about invalidating the cache, the consumtion got changed every billing cycle so this may lead to outdated data!